app/frontend/src/store/tasks.ts

```diff
@@ -30,7 +34,9 @@ $tasks.on(createTaskEffectFx.done, (state, { result }) => [result.task, ...state
  * Updates task in tasks list
  */
 $tasks.on(updateTaskEffectFx.done, (state, { result }) => {
+  // Find index of the task
   const index = state.findIndex(item => result && result.task && item._id === result.task._id);
+  // Replace task at found index with new data
   const newState = [ ...state ].map((item, i) => i === index && result.task ? result.task : item);
```

Comment:
Is it ok that you will change (overwrite) all the state (including all the tasks) when you need to update just a single one?

Reply:
i thought that a reducer should always return new object instead of modifying current state